### PR TITLE
Add github workflows for integration tests

### PR DESCRIPTION
**Pull request by didovesei**
There could have been a more parameterized workflow definition using matrix with nested parameters. I personally prefer this one since it is more straight forwarded and would be easier to maintain...

### CODE REVIEW

File: bin/setup_base [hunk 1]
```diff
@@ -15,7 +21,7 @@ fi
 
 pip3 install -r etc/requirements.txt
 
-DEF_IFACE=`route -n | fgrep UG | awk '{print $8}'`
+DEF_IFACE=`route -n | grep -E '^0\.0\.0\.0' | awk '{print $8}'`
```
Comment (grafnu) [inline bin/setup_base:24]: Use 'egrep \sUG\s' -- Haoli had to do this on another PR as well, so just want to have a consistent solution. @henry54809

Comment (didovesei) [reply bin/setup_base]: Updated.
